### ISSUE TITLE
remove extra margins for pdfs?

### ISSUE THREAD

**Issue by somesortofrescuetoaster** [foobnix/LibreraReader#1232]
Is there a setting I'm missing to get rid of the huge extra margins that librera adds to pdfs?   I hate it so much and makes the app unusable for me trying to read some pdfs.  Scroll lock is not adequate as I sometimes want to pan left and right when zoomed in, but why can't it just stop scrolling at the left and right pdf margins?

**Comment by nobkd** [comment]
I think, that are the default margins.
Yes, there is an option to crop them:

Open a PDF:
![image](https://github.com/foobnix/LibreraReader/assets/44443899/7a826baa-eb1f-4d73-9453-a92424b63c3b)

Get the toolbar:

![image](https://github.com/foobnix/LibreraReader/assets/44443899/39c5baa5-9ef3-465e-8549-2411e62ef387)

Crop area selection:

![image](https://github.com/foobnix/LibreraReader/assets/44443899/dead8f5b-0960-4525-8e40-4e577f50d816)

Maybe change crop setting to your needs manually.

This should work for scroll, as well as book mode. You might need to reopen the PDF after enabling the setting.

Or did I misunderstand something?

**Comment by somesortofrescuetoaster** [comment]
I tried the cropping function.  It doesn't do anything for the extra space that librera seems to put as a background for the pdf?
![Screenshot_20240121-131018](https://github.com/foobnix/LibreraReader/assets/72427461/48378db0-c324-4239-9bca-c6df8c323ee0)
![Screenshot_20240121-131033](https://github.com/foobnix/LibreraReader/assets/72427461/d689d924-db02-4619-838a-753f9c24d2ea)
![Screenshot_20240121-131044](https://github.com/foobnix/LibreraReader/assets/72427461/4d924643-a607-449a-8b0e-1ade07ae8ad6)
![Screenshot_20240121-131120](https://github.com/foobnix/LibreraReader/assets/72427461/0a4773eb-e72a-41d0-8cb7-c687e3b00164)

**Comment by nobkd** [comment]
Ooh. You want to have the panning stop at the edges of the real page? So the page edges are your movement area. 

I personally don't know of any option for that. Sorry.

**Comment by TheHardew** [comment]
That's what the lock in the bottom right corner is for.

**Comment by somesortofrescuetoaster** [comment]
If it's a double column article, the lock button is inconvenient.  I want to zoom in to have a single column for the width, but then when I scroll right, it's too easy to scroll past the margins of the pdf.  Or even if single column, on my small phone, I want to zoom in and scroll left and right without the app going way past the pdf margins.  Probably with better finger dexterity I could make it work fine, but as of now, I find it an annoying experience on my phone compared to other pdf viewers that stop at the pdf margins.

**Comment by nobkd** [comment]
If it's two columns, you can try turning on half-page mode together with locking.

![image](https://github.com/foobnix/LibreraReader/assets/44443899/1569f3d1-fc74-4602-a030-0e87c8709178)
![image](https://github.com/foobnix/LibreraReader/assets/44443899/a3bd4b32-5584-484c-a479-0bf6aa992b0a)


It might help with the size, even, if it's just a little...

**Comment by d3rpd3rpd3rp** [comment]
It would be nice to have a "stay within page margins" feature for those who don't use the page lock (for whatever reason).

**Comment by AlienTux** [comment]
I agree this is most annoying. A setting to stop panning at page margin would be great.
